src/common/error/error-response-body.ts

```diff
@@ -1,4 +1,12 @@
 export default interface ErrorResponseBody {
```

Comment:
You're right. At the moment, we need three definitions - Storefront API, Internal API, and Payment API error responses. I think when the current definition was written, we only considered Internal API.

AFAIK, Storefront API error response is in the following format:

```ts
interface StorefrontErrorResponseBody {
    title: string;
    type: string;
    status: number;
    detail?: string;
    instance?: string;
}
```
It may contain additional properties depending on the endpoint, i.e.: `code` from the coupons endpoint.

The internal API error response is the following:

```ts
interface InternalErrorResponseBody {
    title: string;
    status: number;
    type: string;
    code?: int;
    detail?: string;
    instance?: string;
    errors: string[] | { [key: string]: string };
}
```

Payment API error response is the following:

```ts
interface PaymentErrorResponseBody {
    status: string;
    errors: Array<{ code: string, message: string }>;
}
```

So I'm thinking maybe we can do a union type to represent the variants? i.e.: `type ErrorResponseBody = StorefrontErrorResponseBody | InternalErrorResponseBody | PaymentErrorResponseBody`?

Reply:
yeah I was definitely thinking in doing that, but also thought in doing that change along with changes to RequestError, which at the moment seems to handle way too many scenarios (error message handling)